core/runtime.rs

```diff
@@ -1913,6 +1899,71 @@ pub mod tests {
     });
   }
 
+  #[test]
+  fn overflow_res_async_combined_with_unref() {
```

Comment:
I verified that this test fails on master

Reply:
I did rerun to fix the tests